File: advent-of-code/src/main/java/com/willmolloy/adventofcode/common/grid/MutableCharGrid.java
package com.willmolloy.adventofcode.common.grid;

/**
 * Mutable {@link CharGrid}.
 *
 * @author <a href=https://willmolloy.com>Will Molloy</a>
 */
public final class MutableCharGrid extends CharGrid {
  private final char[][] array;

  public MutableCharGrid(char[][] array) {
    super(array);
    this.array = array;
  }

  /** Set value at a given point. */
  public void set(Point point, char c) throws ArrayIndexOutOfBoundsException {
    array[(int) point.y()][(int) point.x()] = c;
  }
}

File: advent-of-code/src/main/kotlin/com/willmolloy/adventofcode/common/grid/MutableCharGrid.kt
package com.willmolloy.adventofcode.common.grid

/**
 * Mutable [CharGrid].
 *
 * @author <a href=https://willmolloy.com>Will Molloy</a>
 */
class MutableCharGrid(private val array: Array<CharArray>) : CharGrid(array) {

  /** Set value at a given point. */
  @Throws(ArrayIndexOutOfBoundsException::class)
  operator fun set(point: Point, c: Char) {
    array[point.y.toInt()][point.x.toInt()] = c
  }
}
